### DISCUSSION TITLE
[FEATURE REQUEST]: DataFrame to IDataReader

### DISCUSSION THREAD

**Discussion by mikeTWC1984** [dotnet/spark#545] [Q&A]
**Is your feature request related to a problem? Please describe.**

Seems like spark could be  used as JDBC/ADO.NET gateway. At a glance it looks doable to convert DF/IEnumerable to IDataReader auto-magically (and avoid manual type conversion). Any thought? 

**Describe alternatives you've considered**
IKVM

**Additional context**

**Comment by imback82** [comment]
I guess you could wrap Spark's DF in IDataReader's implementation. One thing to note is that Spark's DF is not materialized until you perform an action (such as `Collect`) on it, and `Collect` will bring all the data to the driver. If you are already know this behavior/constraint, you could write one. But I think this can be done outside this repo.

**Comment by mikeTWC1984** [comment]
OK, make sense. Related question, can this project be used as a general purpose .net/java gateway? Like is there a straight forward way to invoke some java class method from spark .net? E.g. ```System.out.println("hello world")```

**Comment by imback82** [comment]
I wouldn't recommend using spark .net for a general purpose .net/java interop. Have you checked something like [jni4net](http://jni4net.com/) (although it hasn't been updated recently)?
